scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -4,122 +4,557 @@
+{
+    { x = 77, y = 0, z = 460 },
+    { x = 71, y = 0, z = 459 },
+    { x = 66, y = 0, z = 458 },
+    { x = 63, y = 0, z = 456 },
+    { x = 60, y = 0, z = 450 },
+    { x = 60, y = 0, z = 443 },
+    { x = 60, y = 0, z = 419 },
+    { x = 40, y = 0, z = 419 },
+    { x = 21, y = 0, z = 419 },
+    { x = -9, y = 0, z = 420 },
+    { x = -59, y = 0, z = 420 },
+    { x = -60, y = 0, z = 433 },
+    { x = -60, y = 0, z = 450 },
+    { x = -62, y = 0, z = 455 },
+    { x = -67, y = 0, z = 458 },
+    { x = -75, y = 0, z = 460 },
+    { x = -76, y = 0, z = 460 },
 }
 
+local middleBlueToFirstBlue = -- Path from the middle blue door to the first blue door
+{
+    { x = -82, y = 0, z = 460 },
+    { x = -93, y = 0, z = 459 },
+    { x = -103, y = 0, z = 459 },
+    { x = -106, y = 0, z = 458 },
+    { x = -110, y = 0, z = 454 },
+    { x = -116, y = 0, z = 434 },
+    { x = -117, y = 0, z = 420 },
+    { x = -121, y = 0, z = 409 },
+    { x = -125, y = 0, z = 395 },
+    { x = -129, y = 0, z = 385 },
+    { x = -133, y = 0, z = 381 },
+    { x = -141, y = 0, z = 380 },
+    { x = -153, y = 0, z = 379 },
+    { x = -157, y = 0, z = 379 },-- Middle blue door
+
+}
 entity.onMobSpawn = function(mob)
+    local pos = mob:getPos()
     mob:setMod(xi.mod.REGAIN, 200)
-    mob:pathThrough(pathNodes, xi.path.flag.PATROL)
+-- Checks for onSpawn to determine which door to go total
+-- Zip popped on blue short side (!pos -196, 0, 396)
+    if pos.x <= -196 and
+        pos.y >= 0 and
+        pos.z >= 396
+    then
+        --print("Zip spawned at Blue Short Side Spawn Point- going to First blue door") useful for debugging if zip gets lost somehow.
+        mob:pathThrough(blueShortSideSpawnPath, xi.path.flag.COORDS)
+    end
+-- Zip popped at the blue basement spawn point (!pos -121, 16, 420)
+    if
+        pos.x <= -121 and
+        pos.y >= 16 and
+        pos.z >= 420
+    then
+        --print(" Zip spawned at Blue basement Spawn Point- going to First blue door") useful for debugging if zip gets lost somehow.
+        mob:pathThrough(blueBasementSpawnPath, xi.path.flag.COORDS)
+    end
+-- Zip popped at the Yellow Short Side Spawn Point - going to first yellow door (!pos 191, 0, 399)
+    if
+        pos.x >= 191 and
+        pos.y >= 0 and
+        pos.z >= 399
+    then
+        --print("Zip spawned at Yellow Short Side Spawn Point- going to First Yellow door", pos) useful for debugging if zip gets lost somehow.
+        mob:pathThrough(yellowShortSideSpawnPath, xi.path.flag.COORDS)
+    end
+-- Zip popped in the yellow side basement - going to first yellow door
+    if
+        pos.x >= 141 and
+        pos.y >= 16 and
+        pos.z >= 440
+    then
+        --print("Zip spawned at Yellow basement Spawn Point- going to First Yellow door", pos) useful for debugging if zip gets lost somehow.
+        mob:pathThrough(yellowBasementSpawnPath, xi.path.flag.COORDS)
+    end
+
+end
```

Comment:
Shouldnt you be using `elseif` for this ones instead of ifs?

Also, there are only 4 options where it can spawn, all of them with severely diferent X positions. Can you check if this keeps working if you simplify the checks to only check for the X position?

Reply:
End result would look like this
```lua
    -- Zip popped on blue short side (!pos -196, 0, 396)
    if pos.x <= -196 then
        --print("Zip spawned at Blue Short Side Spawn Point- going to First blue door") useful for debugging if zip gets lost somehow.
        mob:pathThrough(blueShortSideSpawnPath, xi.path.flag.COORDS)

    -- Zip popped at the blue basement spawn point (!pos -121, 16, 420)
    elseif pos.x <= -121 then
        --print(" Zip spawned at Blue basement Spawn Point- going to First blue door") useful for debugging if zip gets lost somehow.
        mob:pathThrough(blueBasementSpawnPath, xi.path.flag.COORDS)

    -- Zip popped at the Yellow Short Side Spawn Point - going to first yellow door (!pos 191, 0, 399)
    elseif pos.x >= 191 then
        --print("Zip spawned at Yellow Short Side Spawn Point- going to First Yellow door", pos) useful for debugging if zip gets lost somehow.
        mob:pathThrough(yellowShortSideSpawnPath, xi.path.flag.COORDS)

    -- Zip popped in the yellow side basement - going to first yellow door
    elseif pos.x >= 141 then
        --print("Zip spawned at Yellow basement Spawn Point- going to First Yellow door", pos) useful for debugging if zip gets lost somehow.
        mob:pathThrough(yellowBasementSpawnPath, xi.path.flag.COORDS)
    end

Reply:
Ill give it a go - I think we can get away without the Z check but the Y will be needed I would imagine.

Reply:
Looks like I was wrong - seems to be just fine with the x -

Reply:
Helping Hook out with this - So will comment on some of this on his behalf.
Removed all of these `pos.x` checks in lieu of `checkDistance` calls using a table for the spawn points now listed at top of file.